core/src/main/java/com/radixdlt/environment/Dispatchers.java

```diff
@@ -161,19 +158,35 @@ private static final class RemoteDispatcherProvider<T extends RemoteEvent>
     @Override
     public RemoteEventDispatcher<NodeId, T> get() {
       var remoteDispatcher = environmentProvider.get().getRemoteDispatcher(messageType);
-      var localDispatcher = environmentProvider.get().getDispatcher(messageType);
-      var onDispatch =
-          onDispatchProcessors.stream()
-              .flatMap(p -> p.getProcessor(messageType).stream())
-              .collect(Collectors.toSet());
-      return (node, e) -> {
-        if (node.equals(self)) {
-          localDispatcher.dispatch(e);
-        } else {
-          remoteDispatcher.dispatch(node, e);
-        }
-        onDispatch.forEach(p -> p.process(e));
-      };
+
+      if (LocalEvent.class.isAssignableFrom(messageType)) {
+        @SuppressWarnings({"unchecked", "rawtypes"})
+        var localMessageType = (Class<LocalEvent>) (Class) messageType;
+        var localDispatcher = environmentProvider.get().getDispatcher(localMessageType);
+        var onDispatch =
```

Comment:
hmh, it might be better to move this inside `if (node.equals(self)) {` block. We shouldn't  have any "on dispatch" processors for remote events

Reply:
Fixed.